app/main.py

```diff
@@ -1 +1,40 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def die(self) -> None:
+        if self.health <= 0:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        if self.hidden is True:
+            self.hidden = False
+        else:
+            self.hidden = True
```

Comment:
![image](https://github.com/mate-academy/py-herbivores-and-carnivores/assets/93856423/2b13af94-483d-45f6-a7f5-b2f97f7efc41)

please read [checklist.md](https://github.com/mate-academy/py-herbivores-and-carnivores/blob/master/checklist.md#code-efficiency)